src/Compilers/Core/CodeAnalysisTest/PEWriter/BlobTests.cs

```diff
@@ -856,5 +856,31 @@ public void Pooled()
             builder1.Free();
             builder2.Free();
         }
+
+        [Fact]
+        public void ProperStreamRead()
+        {
+            var firstRead = true;
+            var sourceArray = new byte[] { 1, 2, 3, 4 };
+            int sourceOffset = 0;
+
+            var stream = new TestStream(readFunc: (buf, offset, count) =>
+            {
+                if (firstRead)
+                {
+                    count = count / 2;
+                    firstRead = false;
+                }
+                Array.Copy(sourceArray, sourceOffset, buf, offset, count);
+                sourceOffset += count;
+                return count;
```

Comment:
Consider adding a test for scenario when the second read returns 0, i.e. unexpected end of stream.